code/modules/mob/living/basic/farm_animals/bee/_bee.dm

```diff
@@ -0,0 +1,338 @@
+	. = ..()
+	add_traits(list(TRAIT_SPACEWALK, TRAIT_VENTCRAWLER_ALWAYS), INNATE_TRAIT)
+	generate_bee_visuals()
+	AddElement(/datum/element/simple_flying)
+	AddComponent(/datum/component/clickbox, x_offset = -2, y_offset = -2)
+	AddComponent(/datum/component/swarming)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	RegisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET, PROC_REF(apply_chemicals))
+
+/mob/living/basic/bee/mob_pickup(mob/living/picker)
+	if(flags_1 & HOLOGRAM_1)
+		return
+	var/obj/item/clothing/head/mob_holder/destructible/holder = new(get_turf(src), src, held_state, head_icon, held_lh, held_rh, worn_slot_flags)
+	var/list/reee = list(/datum/reagent/consumable/nutriment/vitamin = 5)
+	if(beegent)
+		reee[beegent.type] = 5
+	holder.AddComponent(/datum/component/edible, reee, null, BEE_FOODGROUPS, 10, 0, list("bee"), null, 10)
+	picker.visible_message(span_warning("[picker] scoops up [src]!"))
+	picker.put_in_hands(holder)
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/UnarmedAttack(atom/target, proximity_flag, list/modifiers)
+	if(istype(target, /obj/machinery/hydroponics))
+		var/obj/machinery/hydroponics/hydro = target
+		pollinate(hydro)
+		return
+	if(istype(target, /obj/structure/beebox))
+		var/obj/structure/beebox/hive = target
+		handle_habitation(hive)
+		return
+	return ..()
+
+/mob/living/basic/bee/proc/handle_habitation(obj/structure/beebox/hive)
+	if(hive == beehome) //if its our home, we enter or exit it
+		var/drop_location = (src in beehome.contents) ? get_turf(beehome) : beehome
+		forceMove(drop_location)
+		return
+	if(!isnull(hive.queen_bee) && isqueen) //if we are queen and house already have a queen, dont inhabit
+		return
+	if(!hive.habitable(src) || !isnull(beehome)) //if not habitable or we alrdy have a home
+		return
+	beehome = hive
+	beehome.bees += src
+	if(isqueen)
+		beehome.queen_bee = src
+	return
+
+/mob/living/basic/bee/will_escape_storage()
+	return TRUE
+
+/mob/living/basic/bee/examine(mob/user)
+	. = ..()
+
+	if(!beehome)
+		. += span_warning("This bee is homeless!")
+
+/mob/living/basic/bee/Destroy()
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	if((flags_1 & HOLOGRAM_1))
+		return ..()
+	new /obj/item/trash/bee(loc, src)
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/proc/apply_chemicals(mob/living/basic/attacker, atom/target)
+	SIGNAL_HANDLER
+	if(!beegent || !isliving(target))
+		return
+	var/mob/living/victim = target
+	if(victim.reagents)
+		beegent.expose_mob(victim, INJECT)
+		victim.reagents.add_reagent(beegent.type, rand(1,5))
+
+/mob/living/basic/bee/proc/reagent_incompatible(mob/living/basic/bee/ruler)
+	if(!ruler)
+		return FALSE
+	if(ruler.beegent?.type != beegent?.type)
```

Comment:
can we do an early return if we have no beegent? just `isnull(beegent)`, given how often we access it.

Reply:
the bee can entered the hive even if he dont have a beegent

Reply:
This is `reagent_incompatible()` though? what's that got to do with this?

Reply:
oh you changed it in the code nvm then